tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -1938,12 +1978,18 @@ def _test_enable_fast_capacity_failover(
     clustermgtd_conf_path,
     static_nodes_in_ice_compute_resource,
     ice_dynamic_nodes,
+    target_compute_resource,
 ):
     # set insufficient_capacity_timeout to 180 seconds to quicker reset compute resources
     _set_insufficient_capacity_timeout(remote_command_executor, 180, clustermgtd_conf_path)
-    # trigger insufficient capacity
+
+    # clear slurm_resume and clustermgtd logs in order to start from a clean state
+    remote_command_executor.run_remote_command("sudo truncate -s 0 /var/log/parallelcluster/slurm_resume.log")
+    remote_command_executor.run_remote_command("sudo truncate -s 0 /var/log/parallelcluster/clustermgtd")
+
+    # trigger insufficient capacity: we are using `prefer` to allow requeuing the job on a different CR
     job_id = scheduler_commands.submit_command_and_assert_job_accepted(
-        submit_command_args={"command": "sleep 30", "nodes": 2}
+        submit_command_args={"command": "sleep 30", "nodes": 2, "prefer": target_compute_resource}
```

Comment:
why this require different submission parameter from the ones used in https://github.com/aws/aws-parallelcluster/pull/5381/files#diff-5d2b210a3df82d89d8bed3f1d7d0bc48e5998e3eb9358c07aaa75c5df2d5d44eR1930?

Reply:
It took me a while to figure out.

The first job is expected to succeed ..and for this to happen some the failed nodes must be requeued into another CR ..and for this we need to use `prefer` that "attempt to" follow the constraint ..but falls back to other CRs

The second jobs instead is expected to fail ( `--no-requeue` ) ... `constraint` here is needed to target the expected CR (not sure if it may work also with prefer ...but I used constraint to make the intent of the test slightly more clear.